core/platform/email/gae_email_services_test.py

```diff
@@ -36,10 +45,170 @@ def test_sending_email(self):
             to=feconf.ADMIN_EMAIL_ADDRESS)
         self.assertEqual(0, len(messages))
 
+        messages = self.mail_stub.get_sent_messages(
+            to=feconf.ADMIN_EMAIL_ADDRESS)
+        self.assertEqual(0, len(messages))
         with self.swap(feconf, 'CAN_SEND_EMAILS', True):
             gae_email_services.send_mail(
-                feconf.SYSTEM_EMAIL_ADDRESS, feconf.ADMIN_EMAIL_ADDRESS,
+                feconf.SYSTEM_EMAIL_ADDRESS,
+                feconf.ADMIN_EMAIL_ADDRESS,
                 'subject', 'body', 'html', bcc_admin=False)
         messages = self.mail_stub.get_sent_messages(
             to=feconf.ADMIN_EMAIL_ADDRESS)
         self.assertEqual(1, len(messages))
+
+    def test_that_email_bcc_is_sent_if_bcc_admin_is_true(self):
+        with self.swap(feconf, 'CAN_SEND_EMAILS', True):
+            gae_email_services.send_mail(
+                feconf.SYSTEM_EMAIL_ADDRESS, self.RECIPIENT_EMAIL,
+                'subject', 'body', 'html', bcc_admin=True)
+        messages = self.mail_stub.get_sent_messages(
+            to=self.RECIPIENT_EMAIL)
+        self.assertEqual(1, len(messages))
+        self.assertEqual(messages[0].bcc, feconf.ADMIN_EMAIL_ADDRESS)
+
+    def test_that_email_bcc_not_sent_if_bcc_admin_is_false(self):
+        with self.swap(feconf, 'CAN_SEND_EMAILS', True):
+            gae_email_services.send_mail(
+                feconf.SYSTEM_EMAIL_ADDRESS, self.RECIPIENT_EMAIL,
+                'subject', 'body', 'html', bcc_admin=False)
+        messages = self.mail_stub.get_sent_messages(to=self.RECIPIENT_EMAIL)
+        self.assertEqual(1, len(messages))
+        self.assertFalse(hasattr(messages[0], 'bcc'))
+
+    def test_sending_email_with_reply_to_id_adds_reply_to_email(self):
+        reply_to_id = 'reply_to_id'
+
+        # Tests that email has reply_to_id if passed.
+        with self.swap(feconf, 'CAN_SEND_EMAILS', True):
+            gae_email_services.send_mail(
+                feconf.SYSTEM_EMAIL_ADDRESS, feconf.ADMIN_EMAIL_ADDRESS,
+                'subject', 'body', 'html', reply_to_id=reply_to_id)
+        messages = self.mail_stub.get_sent_messages(
+            to=feconf.ADMIN_EMAIL_ADDRESS)
+        expected_reply_to_address = 'reply+reply_to_id@example.com'
+        self.assertEqual(messages[0].reply_to, expected_reply_to_address)
+
+    def test_sending_email_without_reply_to_id_not_add_reply_to_email(self):
+        # Tests that email does not have reply_to_id if not passed.
```

Comment:
*reply_to email if reply_to_id is not passed.

Reply:
Done.
Thanks!